src/Microsoft.AspNetCore.Server.IISIntegration/Server/IISHttpRequestBody.cs

```diff
@@ -39,7 +39,9 @@ public override int Read(byte[] buffer, int offset, int count)
 
         public override unsafe Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            return _httpContext.ReadAsync(buffer, offset, count, cancellationToken);
+            var memory = new Memory<byte>(buffer, offset, count);
```

Comment:
So what does happen when buffer is null or count is zero?

Reply:
We allow it. After chatting with @davidfowl, a use case could be that someone calls ReadAsync without a buffer to know when data is available. This reduces the amount of idle allocated memory.

Reply:
Does it actually work? You removed the tests.

Reply:
Yeah, I'll need to verify this doesn't null ref.

Reply:
Why would it null ref?

Reply:
Memory supports null buffer, it'll return empty span.